src/lotus/types.rs

```diff
@@ -0,0 +1,196 @@
+use cid::Cid;
+use fvm_shared::address::Address;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::MethodNum;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use strum::{AsRefStr, Display, EnumString};
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct ExecParams {
+    pub code_cid: Cid,
+    pub constructor_params: Vec<u8>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct InstallActorParams {
+    pub code: Vec<u8>,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq, Eq, Display, EnumString, AsRefStr)]
+pub enum WalletKeyType {
+    #[strum(serialize = "bls")]
+    BLS,
+    #[strum(serialize = "secp256k1")]
+    Secp256k1,
+    #[strum(serialize = "secp256k1-ledger")]
+    Secp256k1Ledger,
+}
+
+pub type WalletListResponse = Vec<String>;
+
+/// Helper struct to interact with lotus node
+#[derive(Deserialize, Serialize, Debug)]
+pub struct CIDMap {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "/")]
+    pub cid: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct StateWaitMsgResponse {
+    #[allow(dead_code)]
+    message: CIDMap,
+    #[allow(dead_code)]
+    receipt: Receipt,
+    #[allow(dead_code)]
+    tip_set: Vec<CIDMap>,
+    #[allow(dead_code)]
+    height: u64,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct ReadStateResponse<State> {
+    #[allow(dead_code)]
+    pub balance: String,
+    #[allow(dead_code)]
+    pub code: CIDMap,
+    #[allow(dead_code)]
+    pub state: State,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct Receipt {
+    #[allow(dead_code)]
+    exit_code: u32,
+    #[allow(dead_code)]
+    r#return: String,
+    #[allow(dead_code)]
+    gas_used: u64,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "PascalCase")]
+pub struct MpoolPushMessageResponse {
+    pub message: MpoolPushMessageInner,
+    #[serde(rename = "CID")]
+    pub cid: CIDMap,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "PascalCase")]
+pub struct MpoolPushMessageInner {
```

Comment:
This naming is very confusing. This is actually the meat of the response to `MpoolPushMessage` but this makes it sound like it's something else. I strongly suggest you namespace the types declared in this file so it can be clearly understood what structs are requests, which are responses, etc.